python/equistore-operations/equistore/operations/join.py

```diff
@@ -17,9 +18,42 @@ def join(tensors: List[TensorMap], axis: str):
     `properties` dimension and for ``axis='samples'`` they will be the along the
     samples dimension.
 
-    ``join`` will create an additional label `tensor` specifiying the original index in
-    the list of `tensor_maps`.  If `sample`/`property` names are not the same in all
-    `tensor_maps` they will be unified with a general name ``"property"``.
+    Possible clashes of the meta data like samples/properties names are resolved by one
+    of the three following strategies:
+
+    1. If Labels names are the same and the values are unique, we keep the names and
+       join the values::
+
+        labels_1 = ["n"], [0, 2, 3]
+        labels_2 = ["n"], [1, 4, 5]
+
+       will lead to::
+
+        joined_labels = ["n"],
+                        [0, 2, 3, 1, 4, 5]
+
+    2. If Labels names are the same but the values are not unique, a new dimension
+       ``"tensor"`` is added to the names::
+
+        labels_1 = ["n"], [0, 2, 3]
+        labels_2 = ["n"], [0, 2]
+
+       will lead to::
+
+            joined_labels = ["tensor", "n"],
+                            [[0, 0], [0, 2], [0, 3], [1, 0], [1, 2]]
+
+    3. If Labels names are different we change the names to ("tensor", "property"). This
+       case is only supposed to happen when joining in the property dimension, hence the
+       choice of names::
+
+            labels_1 = ["a"], [0, 2, 3]
+            labels_2 = ["b", "c"], [[0, 0], [1, 2]]
+
+       will lead to::
+
+            joined_labels = ["tensor", "property"],
```

Comment:
What do you think about using a notation for "private" variables in python for metadata column names that can exclusively only used equistore for operations "_eqs_tensor" and "_eqs_property". To be honest, I don't have any other ideas of applications than this function, but it could be already here useful.

Reply:
Hmm I think private names will make everything harder to follow.

Reply:
They actually do this in pandas for their merge function
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.merge.html
> **indicator: bool or str, default False**
>
> If True, **adds a column to the output DataFrame called “_merge” with information on the source of each row**. The column can be given a different name by providing a string argument. The column will have a Categorical type with the value of “left_only” for observations whose merge key only appears in the left DataFrame, “right_only” for observations whose merge key only appears in the right DataFrame, and “both” if the observation’s merge key is found in both DataFrames.